spec/project_spec.rb

```diff
@@ -1029,6 +1040,17 @@ class InvalidErrorHandler; end
     user_id = 'test_user'
     user_attributes = {}
 
+    it 'should return nil when called with invalid project config' do
+      logger = double('logger')
+      allow(logger).to receive(:log)
+      allow(Optimizely::SimpleLogger).to receive(:new) { logger }
+      invalid_project = Optimizely::Project.new('invalid', nil, spy_logger)
+      expect(invalid_project.get_feature_variable_boolean('boolean_single_variable_feature', 'boolean_variable', user_id, user_attributes))
+        .to eq(nil)
+      expect(logger).to have_received(:log).once.with(Logger::ERROR, 'Provided datafile is in an invalid format.')
```

Comment:
Is this right? How come we use similar messages twice?

Reply:
The first one comes from the initialize method. And the second one is logged in the relevant API method. Though duplicate, I think we should keep this at both places for clarity.
If a client is initialized, it should log the datafile error. When an API method is called, it should log exactly why the method is aborting. 
Let me know if we should revise the error log in the API method. And shorten it to only 'Aborting <method name>"

Reply:
Yeah that makes sense. I agree with your assessment.